### PR TITLE
[WIP] SPE-118 Gas migration

### PR DESCRIPTION
**Pull request by couralex6**
# Thank you for contributing to Spellbook!
Please refer to the top of the `readme` in the root of Spellbook to learn how to contribute to Spellbook on DuneSQL.

### CODE REVIEW

**Comment by aalan3** [comment]
@hildobby @MSilb7 these tables are very heavy on the runs. And looking at our internal data the only one that's actually being queried right now is the optimism version. 

How useful are these tables for you at the moment? And are there any plans to build on top of them? Otherwise we are considering not migrating these and removing them except for on Optimism.

**Comment by hildobby** [comment]
> @hildobby @MSilb7 these tables are very heavy on the runs. And looking at our internal data the only one that's actually being queried right now is the optimism version.
> 
> How useful are these tables for you at the moment? And are there any plans to build on top of them? Otherwise we are considering not migrating these and removing them except for on Optimism.

I was thinking of starting to use them soon. Do you think you could migrate them all but disable in prod all of them except optimism? That would make it much easier to re-add them when I want to start using them

Also, this spell came as a follow-up to 3 separate requests from devs for queries on some txs with the same method. I for one want to use it but I'm sure many devs would love to use them (very useful to compare gas optimisation for example). But like most of dune, one of the biggest issues is discoverability and ppl not being aware of the spells that are out there

**Comment by aalan3** [comment]
Sorry I made a mistake 🤦‍♂️ these tables should be migrated without exception. These actually do have some usage. My comment above was meant for `fees_traces` and not `fees`.  I'll add a comment to the discussion and we can continue there. Sorry for the confusion!

**Comment by hildobby** [comment]
> Sorry I made a mistake 🤦‍♂️ these tables should be migrated without exception. These actually do have some usage. My comment above was meant for `fees_traces` and not `fees`. I'll add a comment to the discussion and we can continue there. Sorry for the confusion!

Ah, consider my answer for fees_traces then actually haha

**Comment by aalan3** [comment]
Ah nice 😄 ok then we can exclude the non optimism ones for now and enable them once you start building on top of them again.

**Review by aalan3 — APPROVED** [review]
Could you also add `remove` to the legacy models?

File: models/gas/arbitrum/gas_arbitrum_fees.sql [hunk 1]
```diff
@@ -11,40 +12,44 @@
...
+     cast(NULL as varbinary) as validator, -- Not applicable for L2s
      txns.effective_gas_price/1e9 as gas_price_gwei,
      txns.effective_gas_price/1e18 * p.price as gas_price_usd,
      txns.gas_price/1e9 as gas_price_bid_gwei,
      txns.gas_price/1e18 * p.price as gas_price_bid_usd,
      txns.gas_used as gas_used,
      txns.gas_limit as gas_limit,
-     txns.gas_used / txns.gas_limit * 100 as gas_usage_percent,
+     CASE 
+        WHEN txns.gas_limit = 0 THEN NULL
```
[the remark this answers is not in the document]
Comment (couralex6) [reply models/gas/arbitrum/gas_arbitrum_fees.sql:36]: They're `null` on the spark version right now so I wanted to keep that. We're dividing by 0, so it's not technically 0

[the remark this answers is not in the document]
Comment (aalan3) [reply models/gas/arbitrum/gas_arbitrum_fees.sql:36]: ok that makes sense

**Comment by couralex6** [comment]
Row counts match and the translation for `num_nonzero_bytes` in `gas_optimism_fees` returns the same results

**Pull request merged**